agent/engine/execcmd/manager_start_linux.go

```diff
@@ -135,35 +141,43 @@ func (m *manager) doStartAgent(ctx context.Context, client dockerapi.DockerClien
 	}
 	execRes, err := client.CreateContainerExec(ctx, containerId, execCfg, dockerclient.ContainerExecCreateTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [create]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 
 	seelog.Debugf("Task engine [%s]: created ExecCommandAgent for container: %s -> docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	err = client.StartContainerExec(ctx, execRes.ID, dockerclient.ContainerExecStartTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [pre-start]: %v", err)
+		execMD.Status = apicontainerstatus.ManagedAgentStopped
+		return execMD, StartError{error: err, retryable: true}
 	}
 	seelog.Debugf("Task engine [%s]: sent ExecCommandAgent start signal for container: %s ->  docker exec id: %s", task.Arn, containerId, execRes.ID)
 
 	inspect, err := client.InspectContainerExec(ctx, execRes.ID, dockerclient.ContainerExecInspectTimeout)
 	if err != nil {
-		return StartError{error: err, retryable: true}
+		execMD.StoppedReason = fmt.Sprintf("Unable to start Execute Command Agent [docker inspect]: %v", err)
```

Comment:
`docker inspect` is not very helpful for customers. would suggest removing it.

Reply:
for debug, this tells us where the failure occurred:

`Unable to start Execute Command Agent [create]`
`Unable to start Execute Command Agent [pre-start]`
`Unable to start Execute Command Agent [docker inspect]`

would it be ok to just call it `[inspect]` to differentiate for debug purposes?

Reply:
sgtm